tools/bootstrap-generic-server.py

```diff
@@ -0,0 +1,216 @@
+                          "name": "initScript", "type": "string", "value": f"./{init_script}"})
+            if not os.path.exists(init_script):
+                with open(init_script, "w+") as f:
+                    print(f"""
+#!/usr/bin/env python3
+
+import deviceaccess as da
+import sys
+
+try:
+    da.setDMapFilePath("device.dmap")
+    device = da.Device("{device_name}_unmapped")
+    device.open()
+    # device.write("APP.RESET_N", 1)
+    sys.exit(0)
+except Exception as e:
+    print(e)
+    sys.exit(1)
+""", file=f)
+
+        timer = ET.SubElement(self.root, 'variable', attrib={"name": "periodicTimers", "type": "string"})
+        ET.SubElement(timer, 'value', attrib={"i": str(index), "v": "msTimer"})
+
+        timer_module = ET.SubElement(self.root, 'module', attrib={"name": "msTimer"})
+        ET.SubElement(timer_module, 'variable', attrib={"name": "period", "type": "uint32", "value": "100"})
+
+        if python_modules:
+            py_modules = ET.SubElement(self.root, 'module', attrib={"name": 'PythonModules'})
+
+            for module in python_modules:
+                module_group = ET.SubElement(py_modules, 'module', attrib={"name": module})
+                ET.SubElement(module_group, 'variable', attrib={"name": "path", "type": "string", "value": module})
+
+            # check if user passed us an existing python file, if not: create a simple template
+            if not os.path.exists(module + ".py"):
+                with open(module + ".py", 'w+') as f:
+                    print(f"""
+import PyApplicationCore as ac
+
+class {module}(ac.ApplicationModule):
+    def __init__(self, owner):
+        super().__init__(owner, "{module}", "Please add description here")
+        
+    def mainLoop(self):
+        # Implementation goes here
+        pass
+        
+ac.app.{module} = {module}(ac.app)
+                          """, file=f)
+
+    def write(self):
+        with open('generic_device_server-config.xml', 'w') as f:
+            _pretty_write_xml(self.root, f)
+
+
+class DOOCS:
+    def __init__(self, catalogue, args):
+        self.catalogue = catalogue
+        self.facility, self.device, self.location = args.doocs_address.split("/")
+        self.hostname = args.hostname
+        self.rpc_number = args.doocs_rpc_number
+        self.mapped_device = args.device_name
+
+    def write_config(self):
+        with open('doocs-generic-chimeratk-server02.conf', 'w+') as f:
+            print(f"""
+eq_conf:
+ 
+oper_uid:       -1
+oper_gid:       1000
+xpert_uid:      0
+xpert_gid:      0
+ring_buffer:    10000
+memory_buffer:  500
+ 
+eq_fct_name:    "{self.hostname}._SVR"
+eq_fct_type:    1
+{{
+SVR.RPC_NUMBER:         {self.rpc_number}
+SVR.NAME:       "{self.hostname}._SVR"
+SVR.BPN:        6000
+SVR.FACILITY: "{self.facility}"
+SVR.DEVICE: "{self.device}"
+}}
+eq_fct_name:    "{self.location}"
+eq_fct_type:    10
+{{
+NAME:   "{self.location}"
+}}
+eq_fct_name:    "Debug"
+eq_fct_type:    10
+{{
+NAME:   "Debug"
```

Comment:
It should not be necessary to define any locations, since the Adapter will add them dynamically when missing.

Reply:
Ah right